### PR TITLE
Add SDK files from core repo

### PR DESCRIPTION
**Pull request by delventhalz**
This PR pulls the Javascript SDK files from the core repo , adding them here. This will simplify build processes and the development workflow. In addition to pulling the original files and commits, a number of additional commits rearrange/rename the files and incorporate integration tests.

Note that this is a very large PR with thousands of commits, only the last fifteen of which are original. It is probably simpler to review the [source branch](https://github.com/Cargill/sawtooth-sdk-javascript/tree/c001-add-core-js-sdk) as a final product, then to go commit by commit here.

The sister PR to this one is [Remove JS SDK files from core repo](https://github.com/hyperledger/sawtooth-core/pull/1730) on `hyperledger/sawtooth-core`.

### CODE REVIEW

File: examples/intkey/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6810 @@
+{
```
Comment (peterschwarz) [inline examples/intkey/package-lock.json]: I didn't think this file should be checked in.

Comment (delventhalz) [reply examples/intkey/package-lock.json]: > This file is intended to be committed into source repositories
https://docs.npmjs.com/files/package-lock.json

This goes against the guidance from Node.js, and seems to me to mostly defeat the purpose of a lock file. I disagree entirely. If you are saying it is Sawtooth policy to go against this guidance, then I will remove it in order to comply with that policy.

Comment (peterschwarz) [reply examples/intkey/package-lock.json]: Not checking this in allows us to catch issues that will inevitably crop up during library updates during the CI process.  We should ensure that a nightly build is running for this SDK.

Comment (delventhalz) [reply examples/intkey/package-lock.json]: There is a Jenkinsfile. @rbuysse would know whether or not a nightly build has been set up.

**Comment by dcmiddle** [comment]
@ryjones please override DCO and merge for us.

**Comment by ryjones** [comment]
per request

**Pull request merged**